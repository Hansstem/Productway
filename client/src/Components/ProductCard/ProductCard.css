.productCard {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    gap: 10px;
    align-items: center;
    width: 250px;
    height: 300px;
    margin: 20px;
    padding: 20px;
    border: 1px solid #ccc;
    border-radius: 5px;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
    cursor: pointer;
}

.imgDiv {
    width: 100%;
    height: 70%;
}

.imgDiv img {
    width: 100%;
    height: 100%;
    object-fit: contain;
}

.textDiv {
    width: 100%;
    flex-grow: 1;
    height: auto;
    display: flex;
    flex-direction: column;
    align-items: left;
    justify-content: flex-end;
}
.tittel, .prisParagraph {
    margin: 5px 0px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    color: black;
}


.tittel {
    font-size: 1em;
    line-height: 1.2;
}
.prisParagraph {
    font-size: 0.7em;
}