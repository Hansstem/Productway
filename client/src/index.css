:root {
	font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
	line-height: 1.5;
	font-weight: 400;

	color-scheme: light dark;
	color: #333333;
	background-color: #242424;

	font-synthesis: none;
	text-rendering: optimizeLegibility;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
	-webkit-text-size-adjust: 100%;
}

#root {
	width: 100%;
	max-width: 100%;
	box-sizing: border-box; /* This ensures padding and border are included in the width */
}

a {
	font-weight: 500;
	color: #646cff;
	text-decoration: inherit;
}
a:hover {
	color: #535bf2;
}

body {
	margin: 0;
	display: flex;
	min-height: 100vh;
	background-color: #f5f5f5;
}

h1 {
	font-size: 3.2em;
	line-height: 1.1;
}

button {
	margin: 20px;
	background-color: #f8f8f8;
	color: black;
	border: 1px solid #ddd;
	padding: 10px 20px;
	border-radius: 15px;
	box-shadow: 0px 2px 3px rgba(0, 0, 0, 0.1);
	font-size: 16px;
	transition: all 0.3s ease;
	cursor: pointer;
}

button:hover {
	background-color: #f8f8f8; /* Slightly darker background on hover */
	box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.15); /* More pronounced shadow on hover */
	transform: translateY(-0.5px); /* Slight raise effect on hover */
}

button:active {
	transform: translateY(1px); /* Button appears pressed when active */
}

@media (prefers-color-scheme: light) {
	:root {
		color: #213547;
		background-color: #ffffff;
	}
}
