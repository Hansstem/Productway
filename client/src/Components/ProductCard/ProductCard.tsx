import { useNavigate } from 'react-router-dom';

import { IProduct } from '../../data/types';
import AddToFavorite from '../AddToFavourite/AddToFavourite.tsx';

import './ProductCard.css';

export default function ProductCard({ item }: { item: IProduct }) {
	const navigate = useNavigate();
	const userID = localStorage.getItem('userID') || '';

	return (
		<div className="productCardContainer">
			<div className="favoriteButtonContainer">
				<AddToFavorite
					productID={Number(item.productID)}
					userID={userID}
				/>
			</div>
			<div
				className="productCard"
				onClick={() => navigate('/project2/product/' + item.productID)}
			>
				<div className="imgDiv">
					<img src={item.image} alt={item.name} loading="lazy" />
				</div>
				<div className="textDiv">
					<h6 className="tittel">{item.name}</h6>
					<p className="prisParagraph">
						Pris: <strong>{item.currentPrice} kr</strong> @{' '}
						{item.store}
					</p>
				</div>
			</div>
		</div>
	);
}
