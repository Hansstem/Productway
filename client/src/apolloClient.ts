import { ApolloClient, InMemoryCache } from '@apollo/client';

export const client = new ApolloClient({
	uri: 'http://it2810-10.idi.ntnu.no:4000/graphql',
	credentials: 'include',
	cache: new InMemoryCache(),
});
