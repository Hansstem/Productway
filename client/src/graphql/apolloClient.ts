import { ApolloClient, InMemoryCache } from '@apollo/client';

export const client = new ApolloClient({
	uri: 'http://it2810-10.idi.ntnu.no:4000/graphql', // for production
	//uri: 'http://localhost:4000/graphql', // for local development
	credentials: 'include',
	cache: new InMemoryCache(),
});
