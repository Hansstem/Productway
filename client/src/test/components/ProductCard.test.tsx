import { StrictMode } from 'react';

import { ApolloProvider } from '@apollo/client';
import { render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import ProductCard from '../../components/ProductCard/ProductCard';
import { client } from '../../graphql/apolloClient';

describe('Header test', () => {
	let productCard: RenderResult;
	beforeEach(() => {
		productCard = render(
			<StrictMode>
				<RecoilRoot>
					<ApolloProvider client={client}>
						<MemoryRouter>
							<ProductCard
								item={{
									productID: 1004,
									name: 'Byggryn Hele 550g boks Møllerens',
									brand: 'Møllerens',
									ean: '7020655411993',
									image: 'https://bilder.ngdata.no/7020655411993/kmh/large.jpg',
									store: 'Joker',
									currentPrice: 22.9,
								}}
							/>
						</MemoryRouter>
					</ApolloProvider>
				</RecoilRoot>
			</StrictMode>,
		);
	});

	it('snapshottest productCard', () => {
		expect(productCard).toMatchSnapshot();
	});

	it('renders correctly', () => {
		expect(
			productCard.getAllByText(/Byggryn Hele 550g boks Møllerens/i),
		).toBeTruthy();
		expect(productCard.getAllByText(/Møllerens/i)).toBeTruthy();
		expect(productCard.getAllByText(/Joker/i)).toBeTruthy();
	});
});
