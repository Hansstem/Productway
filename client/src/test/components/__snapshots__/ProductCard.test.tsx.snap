// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header test > snapshottest productCard 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="productCardContainer"
      >
        <div
          class="favoriteButtonContainer"
        >
          <div
            class="addToFavourite"
          >
            <div>
              <div
                class="MuiBox-root css-1b0ve20"
              >
                <button
                  aria-label="Like"
                  class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default css-mm42gz-MuiButtonBase-root-MuiFab-root"
                  data-testid="favoriteIcon"
                  style="background-color: grey; outline: none;"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FavoriteIcon"
                    focusable="false"
                    style="color: white;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          aria-label="Navigate to Byggryn Hele 550g boks Møllerens"
          class="productCard"
        >
          <div
            class="imgDiv"
          >
            <img
              alt="Byggryn Hele 550g boks Møllerens"
              loading="lazy"
              src="https://bilder.ngdata.no/7020655411993/kmh/large.jpg"
            />
          </div>
          <div
            class="textDiv"
          >
            <h6
              class="tittel"
            >
              Byggryn Hele 550g boks Møllerens
            </h6>
            <p
              class="prisParagraph"
            >
              Pris: 
              <strong>
                22.9
                 kr
              </strong>
               @
               
              Joker
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="productCardContainer"
    >
      <div
        class="favoriteButtonContainer"
      >
        <div
          class="addToFavourite"
        >
          <div>
            <div
              class="MuiBox-root css-1b0ve20"
            >
              <button
                aria-label="Like"
                class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default css-mm42gz-MuiButtonBase-root-MuiFab-root"
                data-testid="favoriteIcon"
                style="background-color: grey; outline: none;"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FavoriteIcon"
                  focusable="false"
                  style="color: white;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Navigate to Byggryn Hele 550g boks Møllerens"
        class="productCard"
      >
        <div
          class="imgDiv"
        >
          <img
            alt="Byggryn Hele 550g boks Møllerens"
            loading="lazy"
            src="https://bilder.ngdata.no/7020655411993/kmh/large.jpg"
          />
        </div>
        <div
          class="textDiv"
        >
          <h6
            class="tittel"
          >
            Byggryn Hele 550g boks Møllerens
          </h6>
          <p
            class="prisParagraph"
          >
            Pris: 
            <strong>
              22.9
               kr
            </strong>
             @
             
            Joker
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
